src/server/routes/agent.rs

```diff
@@ -12,6 +12,12 @@ use std::collections::HashMap;
 use std::sync::Arc;
 use warp::{self, Filter, Rejection};
 
+#[derive(Deserialize)]
+pub struct ExperimentData<T> {
+    experiment_name: String,
+    data: T,
```

Comment:
Add the [`#[serde(flatten)]`](https://serde.rs/attr-flatten.html) attribute to `data`. This way the API can be called like:

```json
{
    "experiment-name": "foo",
    "results": [...],
    "shas": [...]
}
```